<?xml version="1.0"?>
<launch>

  <param name="use_sim_time" value="true"/>

  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find liodom)/config/logging.conf" />

  <node pkg="liodom"
        type="liodom_node"
        name="liodom"
        output="screen"
        respawn="false">

    <param name="min_range" type="double" value="3.0" />
    <param name="max_range" type="double" value="75.0" />
    <param name="lidar_type" type="int" value="1" />
    <param name="scan_lines" type="int" value="64" />
    <param name="scan_regions" type="int" value="8" />
    <param name="edges_per_region" type="int" value="10" />
    <param name="prev_frames" type="int" value="15" />
    <param name="save_results" type="bool" value="true" />
    <param name="save_results_dir" type="string" value="/home/emilio/Escritorio/results/" />
    
    <remap from="~points" to="/os1_cloud_node/points" />
  </node>  

  <node pkg="tf"
        type="static_transform_publisher"
	      name="map2odom"
	      args="0 0 0 0 0 0 map odom 1000" >
  </node>

  <node pkg="tf" 
        type="static_transform_publisher" 
	      name="bl2os1"
	      args="0 0 0 0 0 0 base_link os1_lidar 1000" >
  </node>

  <!--node pkg="tf"
        type="static_transform_publisher" 
	      name="os12imu"
	      args="0.006253 -0.011775 0.007645 0 0 0 1 os1_sensor os1_imu 1000" >
  </node>

  <node pkg="tf"
        type="static_transform_publisher" 
	      name="os12lidar"
	      args="0 0 0.03618 0 0 1 0 os1_sensor os1_lidar 1000" >
  </node-->

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find liodom)/rviz/liodom_ouster.rviz" />

</launch>
